java/src/main/java/com/codurance/training/tasks/TaskList.java

```diff
@@ -4,15 +4,21 @@
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.PrintWriter;
-import java.util.ArrayList;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
 
 public final class TaskList implements Runnable {
     private static final String QUIT = "quit";
 
     private final Map<String, List<Task>> tasks = new LinkedHashMap<>();
+
+    private final AddProjectImp addProjectImp;
+    private final UncheckTaskImp uncheckTaskImp;
+    private final CheckTaskImp checkTaskImp;
+
+    private final AddTaskImp addTaskImp;
+    private final TaskView taskView;
```

Comment:
Replace the types of above objects with their interface names... 
like AddProjectImp implements TaskService.. so type of the object should be TaskService instead of class name (to ensure dependency inversion)